javascript/map_utils.js

```diff
@@ -106,263 +39,228 @@ function lonLatToMercator(ll) {
  *
  * @coord {number} - the datum, a latitude or a longitude
  * @format {string} - the format
- * @throws Will throw exceptions if the format string is malformatted. 
+ * @throws Will throw exceptions if the format string is malformatted.
  */
 function formatCoords(coord, format) {
-    function didf(x,di,df,sep,fill) {
-        let s = x.toFixed(df);
-        s = s.replace(/[.,]/,sep);
-        let z = df+di+(df>0?1:0)-s.length;
-        if (z > 0) {
-            /* need to prepend zeros */
-            s = fill.repeat(z) + s;
-        }
-        return s;
+  function didf(x, di, df, sep, fill) {
+    let s = x.toFixed(df);
+    s = s.replace(/[.,]/, sep);
+    let z = df + di + (df > 0 ? 1 : 0) - s.length;
+    if (z > 0) {
+      /* need to prepend zeros */
+      s = fill.repeat(z) + s;
     }
-
-    // OL2 has a bug (?) where sometimes the coordinates go beyond +-180
-    if (coord> 180) coord -= 360;
-    if (coord<-180) coord += 360;
-
-    let a = Math.abs(coord);
-    let deg = Math.trunc(a);
-    let b = 60*(a-deg);
-    let min = Math.trunc(b);
-    let sec = 60*(b-min);
-
-    let s = '';
-    let i=0;
-
-    let di = 0;
-    let df = 0;
-    let sep = '';
-    let fill = '#';
-
-    do {
-        let c = format.charAt(i);
-        switch (c) {
-        case 'N': case 'S':
-            s += (coord >= 0 ? 'N':'S'); i++; sep = '#';
-            break;
-        case 'W': case 'E':
-            s += (coord >= 0 ? 'E':'W'); i++; sep = '#';
-            break;
-        case ' ':
-            s += ' '; i++; sep = '#';
-            break;
-        case '#':
-        case '_':
-            di = 0;
-            df = 0;
-            fill = (c == '_'?' ':'0');
-            do {
-                di++; i++;
-            } while (format.charAt(i) === c);
-            if (format.charAt(i) == ',' || format.charAt(i) == '.') {
-                sep = format.charAt(i); i++;
-            } else {
-                continue;
-            }
-            while (format.charAt(i) === c) {
-                df++; i++;
-            }
-            break;
-        case '°':
-            if (fill === '#') {
-                throw 'missing format specifier';
-            }
-            /* If decimal places are to be rendered, use the full number.
+    return s;
+  }
+
+  // OL2 has a bug (?) where sometimes the coordinates go beyond +-180
```

Comment:
Is this code still necessary or appropriate? It seems to refer to OL2 only.

Reply:
I adapted the comment but I will not change the code. Too scary :D and not time to check.